src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,39 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { getUser } from '../../api';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
+
+interface Props {
+  selectedTodo: Todo;
+  openModal: (todo: null) => void;
+}
+
+export const TodoModal: React.FC<Props> = React.memo(({
+  selectedTodo,
+  openModal,
+}) => {
+  const [user, setUser] = useState<User | null>(null);
+
+  const fetchData = async () => {
+    try {
+      const data = await getUser(selectedTodo.userId);
+
+      setUser(data);
+    } catch {
+      throw new Error('There is an error with loading');
```

Comment:
in this way your app will be broken
you shouldn't throw an error here, will be better to manage some error message with boolean state variable